### PR TITLE
Ejercicio 3 find names

### LINKED ISSUE #3: findNames.sh
Identificación de nombres propios (se identifican sólo si están en este formato Nnnnnnnnn), aunque la palabra no sea un nombre propio realmente. Ejemplos: Mateo, Estonoesunnombre, Ana.

### PR DESCRIPTION
**Pull request by Flor243**

### CODE REVIEW

**Review by salvanya — APPROVED** [review]
The feature works correctly and I'm not able to find bugs

File: findNames.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            if [[ " ${NAMES[*]} " =~ " $WORD " ]]; then
+             continue
+            fi
+                
+            NAMES+=( $WORD )
+        fi
+    done;
+done < $FILE
+
+IFS=$'\n' NAMES_SORTED=($(sort <<<"${NAMES[*]}"))
```
Comment (fedeantuna) [inline findNames.sh:33]:
```suggestion
PREV_IFS=$IFS
IFS=$'\n' NAMES_SORTED=($(sort <<<"${NAMES[*]}"))
```

File: findNames.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+for WORD in ${NAMES_SORTED[@]}
+do
+    echo $WORD
+done
```
Comment (fedeantuna) [inline findNames.sh]:
```suggestion
done

IFS=$PREV_IFS
```

File: findNames.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+do
+    for WORD in "${LINE[@]}";
+    do
+        if [[ $WORD =~ [,.]$ ]]
+        then
+            WORD=${WORD::-1}
+        fi
+ 
+        if [[ $WORD =~ ^[A-ZÄËÏÖÜÁÉÍÓÚÂÊÎÔÛÀÈÌÒÙÑ][a-zäëïöüáéíóúáéíóúâêîôûàèìòùñ]+$ ]]
+        then
```
Comment (fedeantuna) [inline findNames.sh]:
```suggestion
        if [[ $WORD =~ ^[A-ZÄËÏÖÜÁÉÍÓÚÂÊÎÔÛÀÈÌÒÙÑ][a-zäëïöüáéíóúáéíóúâêîôûàèìòùñ]+$ ]]; then
```

File: findNames.sh [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+### Then, we normalize every present word in each line and we use a regex to evaluate if it is an own name.
+### If it is and the current own name is not in the name's array, we added to it.
+### Finally, we sort the array of names and print each element that it has.
+
+
+FILE=$1
+
+NAMES=()
+
+while read -ra LINE;
```
Comment (salvanya) [inline findNames.sh]:
```suggestion
while read -ra LINE
```

File: findNames.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+FILE=$1
+
+NAMES=()
+
+while read -ra LINE; 
+do
+    for WORD in "${LINE[@]}";
+    do
+        if [[ $WORD =~ [,.]$ ]]
+        then
```
Comment (fedeantuna) [inline findNames.sh]:
```suggestion
        if [[ $WORD =~ [,.]$ ]]; then
```

File: findNames.sh [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ 
+        if [[ $WORD =~ ^[A-ZÄËÏÖÜÁÉÍÓÚÂÊÎÔÛÀÈÌÒÙÑ][a-zäëïöüáéíóúáéíóúâêîôûàèìòùñ]+$ ]]
+        then
+            if [[ " ${NAMES[*]} " =~ " $WORD " ]]; then
+             continue
+            fi
+                
+            NAMES+=( $WORD )
+        fi
+    done;
```
Comment (salvanya) [inline findNames.sh]:
```suggestion
    done
```

File: findNames.sh [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+### Finally, we sort the array of names and print each element that it has.
+
+
+FILE=$1
+
+NAMES=()
+
+while read -ra LINE; 
+do
+    for WORD in "${LINE[@]}";
```
Comment (salvanya) [inline findNames.sh]:
```suggestion
    for WORD in "${LINE[@]}"
```

File: findNames.sh [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+while read -ra LINE
+do
+    for WORD in "${LINE[@]}"
+    do
+        if [[ $WORD =~ [,.]$ ]]; then
+            WORD=${WORD::-1}
+        fi
+ 
+        if [[ $WORD =~ ^[A-ZÄËÏÖÜÁÉÍÓÚÂÊÎÔÛÀÈÌÒÙÑ][a-zäëïöüáéíóúáéíóúâêîôûàèìòùñ]+$ ]]; then
+            if [[ " ${NAMES[*]} " =~ " $WORD " ]]; then
```
Comment (fedeantuna) [inline findNames.sh]:
```suggestion
            if [[ "${NAMES[*]}" =~ "$WORD" ]]; then
```

**Review by fedeantuna — APPROVED** [review]

**Review by salvanya — APPROVED** [review]

**Pull request merged**